### PR TITLE
Fix list_job_excecutions_for_thing with status parameter

### PR DESCRIPTION
**Pull request by gofabian**
If the methods `list_job_executions_for_thing` and `list_job_executions_for_job` are used with an explicit `status` parameter then it will always run into a `KeyError "status"`.

[log.txt](https://github.com/spulec/moto/files/7218426/log.txt)

This PR fixes a bug where wrong fields of job executions are accessed when filtering by status.

### CODE REVIEW

File: moto/iot/models.py [hunk 1]
```diff
@@ -1338,7 +1338,7 @@ def list_job_executions_for_thing(
         if status is not None:
             job_executions = list(
                 filter(
-                    lambda elem: status in elem["status"] and elem["status"] == status,
+                    lambda elem: "status" in elem["jobExecutionSummary"] and elem["jobExecutionSummary"]["status"] == status,
```
Comment (bblommers) [inline moto/iot/models.py:1341]:
```suggestion
                    lambda elem: elem["jobExecutionSummary"].get("status") == status
```

Comment (bblommers) [reply moto/iot/models.py:1341]: This maybe a  more readable alternative - logically it does the same as far as I can see

**Comment by gofabian** [comment]
> tests/test_s3/test_s3.py::test_missing_key_request_boto3 FAILED

@bblommers  Any idea what is the problem here? I have no clue why my changes should have influenced that test?

**Comment by bblommers** [comment]
> 
> 
> > tests/test_s3/test_s3.py::test_missing_key_request_boto3 FAILED
> 
> @bblommers Any idea what is the problem here? I have no clue why my changes should have influenced that test?

Hey @gofabian , that looks like an issue with our CI - every now and then the Docker container decides not to respond. I've restarted the tests - should be fine the second time 🤞

**Review by bblommers — APPROVED** [review]
LGTM! Thanks for the fix @gofabian

**Pull request merged**